google_cdn-external/main.tf

```diff
@@ -3,21 +3,26 @@
  */
 
 locals {
-  name_prefix             = join("-", [var.application, var.environment, var.name != "" ? "${var.name}-cdn" : "cdn"])
-  url_map_default_service = var.backend_type == "bucket" ? try(google_compute_backend_bucket.default[0].id, "") : try(google_compute_backend_service.default[0].id, "")
-  url_map_self_link       = var.backend_type == "bucket" ? try(google_compute_backend_bucket.default[0].self_link, "") : try(google_compute_backend_service.default[0].self_link, "")
+  name_prefix = join("-", [var.application, var.environment, var.name != "" ? "${var.name}-cdn" : "cdn"])
+  # when both a bucket and backend service are specified, prefer the backend
+  # service as the default backend, and use backend_bucket_paths to route
+  # specific paths to the backend bucket
+  url_map_default_service        = var.backend_type == "bucket" ? one(google_compute_backend_bucket.default[*].id) : one(google_compute_backend_service.default[*].id)
+  url_map_self_link              = var.backend_type == "bucket" ? one(google_compute_backend_bucket.default[*].self_link) : one(google_compute_backend_service.default[*].self_link)
+  backend_bucket_default_service = one(google_compute_backend_bucket.default[*].id)
```

Comment:
I'm not seeing this referenced anywhere. Do you expect you'll need to use it at some point?

Reply:
Good catch. I will double check but I think I might have meant to reference it in the case where both a service and backend bucket were specified, in which case I'll update it to be referenced. Otherwise I'll remove it.

Reply:
Removed. It's a tad confusing that the `default_service` parameter at https://github.com/mozilla/terraform-modules/blob/main/google_cdn-external/main.tf#L121 and the one at https://github.com/mozilla/terraform-modules/blob/main/google_cdn-external/main.tf#L135 take different representations, but that's a Terraform/GCP thing. In any case for bucket `backend_bucket_paths` only the `self_link` version is needed.

Reply:
Gotcha, makes sense. Approved the PR.

Reply:
whoops looks like I re-requested review after you had already approved, can you re-approve?

Reply:
np, done